### ISSUE TITLE
Inlet and Outlet vector is zero

### ISSUE THREAD

**Issue by luizfuchs** [DualSPHysics/DesignSPHysics#61]
[Testcase-Inlet.zip](https://github.com/DualSPHysics/DesignSPHysics/files/4258408/Testcase-Inlet.zip)
Hello,

I was reading the issues and i found at issue #50 that the Inlet/Outlet problems were solved.

So, I tryied to run a model to test this function (added recentrly to the GUI)
As I am not an expert user I probably made a mistake during the process, so I will wrote my model steps until the error occurs to identify if it was my mistake or a bug on the implementation of the In/Out.

Well, I was trying to replicate the "ShapesInlet3D" problem presented in https://www.youtube.com/watch?v=qcS2BHrMW0U

Modeling Steps:
1) A new model was created with the last version of the software
2) created a floor and an inlet box, to feed the model.
![image](https://user-images.githubusercontent.com/57912027/75391707-5c64c480-58c9-11ea-91d4-ce09004c2c5e.png)
3) The floor was added to the model as a boundary and the box (inlet) as a fluid MK Fluid: 0.
![image](https://user-images.githubusercontent.com/57912027/75392627-304a4300-58cb-11ea-92b2-f09a303f9b23.png)

4) I set the Inlet/Outlet as the following image
![image](https://user-images.githubusercontent.com/57912027/75391811-93d37100-58c9-11ea-8cef-fdebb48b4d51.png)
5) When I run the model, the following message appeared.

> Total particles: 20242 (bound=20000 (fx=20000 mv=0 ft=0) fluid=242)

> Total MK blocks: 2 (bound=1 (fx=1 mv=0 ft=0) fluid=1)
 
> InOut configuration:
> *** Exception (JSphInOutPoints::CheckPoints) at ..\source\JSphInOutPoints.cpp:568
> Text: Direction vector is zero.
> File: F:/OA Engenharia/SPH/Testcase-Inlet/Testcase-Inlet_out/Testcase-Inlet.xml(row:58)

> Finished execution (code=1).

6) I opened the xml file, looking for the row:58
![image](https://user-images.githubusercontent.com/57912027/75392384-ae5a1a00-58ca-11ea-9b30-99abfcffbb96.png)
 7) Row:58 appears to be related to layer.
8) I changed the layer number a few more times but the error still appearing.

Well, my question is.... Is this a mistake that I made or it is a bug? 

P.S.: If this problem was my fault, I will redirect my post to DualPhysiscs forum to discuss modeling problems  https://forums.dual.sphysics.org/categories/dualsphysics-v4-4/p2

P.S.2: I had attached the file if you need more informations

**Comment by zlotny** [comment]
Hi there. Indeed it was an error on my application. For some reason the "zone3d" XML template was empty, although the 2D one works just fine.

This is fixed and added to the master branch, so you can update DesignSPHysics with the addon manager. The commit that fixed this is f4b4971415b0571355d8b1f9a7881387316abfce

Also, now there's another problem:
```
*** Exception (JSphInOutPoints::Create2d3d_Particles) at JSphInOutPoints.cpp:160
Text: Domain size of particles in direction X is not zero.
File: /home/ndrs/Downloads/Testcase-Inlet/Testcase-Inlet_out/Testcase-Inlet.xml(row:59)
```
Though that appears to be already solved in the forums, so check that out: https://forums.dual.sphysics.org/discussion/1627/error-outlet-domain-size-of-particles-in-direction-z-is-not-zero


However that problem is related to DualSPHysics and how to model cases, not inlet configuration not exporting correctly. So I'm marking this as fixed and I'll close the issue. I don't really know how to work with Inlet/Outlet so I can't help much in this new error :sweat_smile: 

Thanks for reporting the issue, this was sneaking out of my sight for a lot of time.

**Comment by luizfuchs** [comment]
Hello again,

I was looking at the model forum and I guess there is another problem...

At the outlet I have tried to set the volicity as an extrapolated value. It means that the XML file had to set the Impose velocity to "2"

![image](https://user-images.githubusercontent.com/57912027/75474691-30534d00-5976-11ea-9f69-fc431dbd4feb.png)

> <imposevelocity mode="2" comment="Imposed velocity 0:fixed value, 1:variable value, 
> 2:Extrapolated velocity, 3:Interpolated velocity (default=0)"> </imposevelocity>

Unfortunately, it is not changed at the xml file, getting back to the dafult value of "0" (meaning fixed velocity value)

![image](https://user-images.githubusercontent.com/57912027/75474466-ce92e300-5975-11ea-97b9-6bdf669e831b.png)

![image](https://user-images.githubusercontent.com/57912027/75474664-26c9e500-5976-11ea-92d2-e6ac0dbd9638.png)

Hope you can solve it (again, as always haha)

**Comment by zlotny** [comment]
Hi there, you're right. That somehow slipped through and I was saving that GUI information on the wrong data structure :sweat_smile: 

It is now fixed (commit a00f30d48a44bbca2fbbb23cb675272453c6c975) and integrated into the master branch, so you can update your version via de Addon Manager.

I also reworked the Inlet / Outlet window a bit becaus it had a weird bug while deleting zones :/

Let's hope this is now right. Closing the issue... again :)
